DbAccessor/src/Shared/Utilities.cs

```diff
@@ -0,0 +1,52 @@
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+
+namespace GoRideShare
+{
+    public static class Utilities
+    {
+        // Method that validates headers, outputs the userID and dbToken, returns exception if headers  missing, null if headers are good
+        public static IActionResult ValidateHeaders(IHeaderDictionary headers, out string userId)
+        {
+            userId = string.Empty;
+            
+            // Check for X-User-ID  and X-DbToken headers
+            if (!headers.TryGetValue("X-User-ID", out var userIdValue) || string.IsNullOrWhiteSpace(userIdValue))
+            {
+                return new BadRequestObjectResult("Missing the following header: 'X-User-ID'.");
+            }
+            userId = userIdValue.ToString();
+
+            return null; // All headers are valid
+        }
+
+        internal static async Task<(bool error, Stream response)> MakeHttpGetRequest(string xUserId, string endpoint)
```

Comment:
Returning a Stream instead of a String doesn't allow us to return an error message on failure, effectivly throwing out the debugging info.

If we instead use something like `ReadToEndAsync()` then we can return a string (either the response or the error message depending on the boolean flag)

I'm not super familiar with Mongo though, so it possible there is no variablity in the error messages already.

Currently we always use 

```
return new ObjectResult("Failed to get user details from DB") { StatusCode = StatusCodes.Status500InternalServerError };
```
Which might be sufficient if there is only one kind of error we can get from the DB.

Reply:
fixed it